src/lib/Libifl/pbsD_stathost.c

```diff
@@ -125,7 +125,7 @@ add_consumable_entry(struct attrl *patl,
 		return;
 
 	for (i = 0; i < *consumable_size; ++i) {
-		if (((*consum) + i) == NULL)
+		if (((*consum) + i)->cons_resource == NULL)
```

Comment:
If (*consum + i) == NULL then this will throw an error right?

Reply:
it will be NULL only when the variable i is zero, and in that case, we will not enter the for loop.
In other functions - the code does not check for consum to be NULL.

https://github.com/openpbs/openpbs/blob/f4103f0ec40aa861e8a8cdc9e8d20dbcd25b3c81/src/lib/Libifl/pbsD_stathost.c#L339
https://github.com/openpbs/openpbs/blob/f4103f0ec40aa861e8a8cdc9e8d20dbcd25b3c81/src/lib/Libifl/pbsD_stathost.c#L691
https://github.com/openpbs/openpbs/blob/f4103f0ec40aa861e8a8cdc9e8d20dbcd25b3c81/src/lib/Libifl/pbsD_stathost.c#L736